src/python/vdo-setup-dmtest/vdo-setup-dmtest.py

```diff
@@ -0,0 +1,512 @@
+        self.config["_".join([repoName.upper(), 'PATH'])] = repoPath
+        self.log.debug("Repo path is: {0}".format(repoPath))
+      elif os.path.exists(self.config[key]):
+        # The key value is a valid path, so proceed accordingly
+        repoPath = self.config[key]
+
+        # Add the repo path to the config structure
+        repoName = repoPath.rpartition("/")[-1]
+        self.config["_".join([repoName.upper(), 'PATH'])] = repoPath
+        self.log.debug("Repo path is: {0}".format(repoPath))
+      else:
+        # The key value is invalid
+        error = "Invalid definition"
+        self.log.error("Failed to process repo key {key}: {reason}".format(key = key,
+                                                                           reason = error))
+        sys.exit(1)
+
+  def installDeps(self):
+    """
+    Install package dependencies
+    """
+    self.log.info("Installing package dependencies listed in config file");
+    if (self.config['PACKAGES']):
+      try:
+        result = runCommand(['sudo dnf install -y', " ".join(self.config['PACKAGES'])])
+        assert result.returncode == 0, "Non-zero return status received"
+      except Exception as error:
+        self.log.error("Failed to install the required dependencies: {0}".format(error))
+        sys.exit(1)
+    else:
+      self.log.info("Skipping - no package dependencies to install");
+
+  def runRepoInstallSteps(self):
+    """
+    Run the correpsonding install method for each repository that was cloned
+    """
+    self.log.info("\nRunning the install methods for each repo")
+    for key in self.repoKeys:
+      repoName = self.config[key].rpartition("/")[-1].rsplit(".", 1)[0]
+      repoPath = self.config["_".join([repoName.upper(), 'PATH'])]
+
+      # Perform necessary installation steps for the repos
+      self.log.info("Installing {0}".format(repoName))
+      try:
+        os.chdir(repoPath)
+
+        # The repo keys are sorted, so install should go in this order
+        if (key.upper().find('BUFIO') != -1):
+          self.installBufioTest(git.Repo(repoPath))
+        elif (key.upper().find('BLK_ARCHIVE') != -1):
+          self.installBlkArchive(repoName)
+        elif (key.upper().find('DMTEST') != -1):
+          self.installDmtestPython(repoName)
+        else:
+          msg = 'does not match any of the expected substrings (BUFIO | BLK_ARCHIVE | DMTEST)'
+          self.log.warning("Repo key '{key}' {error} - Skipping".format(key = key, error = msg))
+      except Exception as error:
+        self.log.error("Failed during {name} installation: {reason}".format(name = repoName,
+                                                                            reason = error))
+        # If dmtest-python install failed, do not continue any further
+        if (key.upper().find('DMTEST') != -1):
+          sys.exit(1)
+        else:
+          self.log.warning("Manual install of {name} required".format(name = repoName))
```

Comment:
Is it worth noting the location that it should be installed to?

Reply:
Fixed in latest push.